### PR TITLE
Do not exclude any linter errors

### PR DESCRIPTION
**Pull request by travisperson**
golangci-lint chooses to ignore a few linter errors by default. One of these was the doc comments from golint which is why we haven't seen these issues fail ci. This PR disables this configuration and doesn't silent any linter errors.

There are a things that still need proper comments, I've noted these with a `TODO(author)`. Comment on the line with whatever I should replace the placeholder with and I'll update the PR.

### CODE REVIEW

File: types/porep_proof_partitions.go [hunk 1]
```diff
@@ -1,8 +1,11 @@
 package types
 
+// PoRepProofPartitions TODO(laser)
```
Comment (laser) [inline types/porep_proof_partitions.go]: PoRepProofPartitions represents the number of partitions used when creating a PoRep proof, and impacts the size of the proof.

File: types/porep_proof_partitions.go [hunk 2]
```diff
@@ -1,8 +1,11 @@
...
 type PoRepProofPartitions uint64
 
 const (
+	// TestPoRepProofPartitions TODO(laser)
```
Comment (laser) [inline types/porep_proof_partitions.go]: TestPoRepProofPartitions is an opaque value signaling that an unknown number of partitions were used when creating a PoRep proof in test mode.

File: types/porep_proof_partitions.go [hunk 3]
```diff
@@ -1,8 +1,11 @@
...
 	TestPoRepProofPartitions = PoRepProofPartitions(iota)
+	// TwoPoRepPartitions TODO(laser)
```
Comment (laser) [inline types/porep_proof_partitions.go]: TwoPoRepPartitions indicates that two partitions were used to create a PoRep proof.

File: types/sector_size.go [hunk 4]
```diff
@@ -6,6 +6,9 @@ package types
 type SectorSize uint64
 
 const (
+	// OneKiBSectorSize TODO(laser)
```
Comment (laser) [inline types/sector_size.go]: OneKiBSectorSize indicates a sector which, after sealing, contains 1024 bytes.

File: types/sector_size.go [hunk 5]
```diff
@@ -6,6 +6,9 @@ package types
...
 	OneKiBSectorSize = SectorSize(iota)
+
+	// TwoHundredFiftySixMiBSectorSize TODO(laser)
```
Comment (laser) [inline types/sector_size.go]: TwoHundredFiftySixMiBSectorSize indicates a sector which, after sealing, contains 256MiB.

File: types/post_proof_partitions.go [hunk 6]
```diff
@@ -1,8 +1,11 @@
 package types
 
+// PoStProofPartitions TODO(laser)
```
Comment (laser) [inline types/post_proof_partitions.go]: PoStProofPartitions represents the number of partitions used when creating a PoSt proof, and impacts the size of the proof.

File: types/post_proof_partitions.go [hunk 7]
```diff
@@ -1,8 +1,11 @@
...
 type PoStProofPartitions uint64
 
 const (
+	// TestPoStPartitions TODO(laser)
```
Comment (laser) [inline types/post_proof_partitions.go]: TestPoStPartitions is an opaque value signaling that an unknown number of partitions were used when creating a PoSt proof in test mode.

File: types/post_proof_partitions.go [hunk 8]
```diff
@@ -1,8 +1,11 @@
...
 	TestPoStPartitions = PoStProofPartitions(iota)
+	// OnePoStPartition TODO(laser)
```
Comment (laser) [inline types/post_proof_partitions.go]: OnePoStPartition indicates that a single partition was used to create a PoSt proof.

File: porcelain/protocol.go [hunk 9]
```diff
@@ -11,6 +11,7 @@ import (
 	"github.com/filecoin-project/go-filecoin/types"
 )
 
+// ProtocolParams TODO(rosa)
```
Comment (rosalinekarr) [inline porcelain/protocol.go]: ProtocolParams are a commonly inspected set of parameters for the storage and retrieval protocols that we make available for the user to inspect via the `go-filecoin protocol` command.

File: porcelain/protocol.go [hunk 10]
```diff
@@ -21,6 +22,7 @@ type protocolParamsPlumbing interface {
 	MessageQuery(ctx context.Context, optFrom, to address.Address, method string, params ...interface{}) ([][]byte, error)
 }
 
+// ProtocolParameters TODO(rosa)
```
Comment (rosalinekarr) [inline porcelain/protocol.go]: ProtocolParameters exposes certain commonly inspected parameters of the storage and retrieval protocols for use by the user and elsewhere in the application.

File: repo/fsrepo_test.go [hunk 11]
```diff
@@ -80,8 +80,9 @@ func TestFSRepoInit(t *testing.T) {
 
 	dir, err := ioutil.TempDir("", "")
 	assert.NoError(t, err)
-
-	defer os.RemoveAll(dir)
+	defer func() {
+		require.NoError(t, os.RemoveAll(dir))
+	}()
```
Comment (anorth) [inline repo/fsrepo_test.go]: It would be worth factoring this out to a function, but you don't have to do that here.

Comment (ZenGround0) [reply repo/fsrepo_test.go]: Along the same lines I have a slight preference for putting this all on one lines but that's purely a style opinion.

File: tools/iptb-plugins/filecoin/local/localfilecoin.go [hunk 12]
```diff
@@ -520,7 +520,7 @@ func (l *Localfilecoin) SwarmAddrs() ([]string, error) {
 
 /** Config Interface **/
 
-// GetConfig returns the nodes config.
+// Config returns the nodes config.
```
Comment (ZenGround0) [inline tools/iptb-plugins/filecoin/local/localfilecoin.go]: super nit: throw an apostrophe in there

File: repo/fsrepo_test.go (hunk 11 above)
Comment (travisperson) [reply repo/fsrepo_test.go]: Ya, I think we can come back and make a good-first-issue to do this work.

**Pull request merged**